modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -112,6 +167,32 @@ trait MAuthDirectives extends StrictLogging {
       None
     }
   }
+
+  private def getQueryString(req: HttpRequest): String =
+    if (req.uri.rawQueryString == None) "" else req.uri.rawQueryString.get
+
+  private def validateAuthenticationValue(headerName: String, authentication: String): Directive1[String] = {
+    val mauthVersion: MAuthVersion = if (headerName eq `MCC-Authentication`.name) MAuthVersion.MWSV2 else MAuthVersion.MWS
```

Comment:
This doesn't feel right, `headerName` is a string and can be anything really, it's not safe currently

Reply:
check if the headerName is MCC-Authentication or X-MWS-Authentication first, see the fix in  https://github.com/mdsol/mauth-jvm-clients/pull/92/commits/bfa021f78a935387f61abda605b5bddcedd1f62a